boa/dealer.py

```diff
@@ -0,0 +1,90 @@
+import boa
+from boa.util.abi import Address
+
+SLOAD_OPCODE = 0x54
+
+
+class SloadTracer:
+    def __init__(self, super_fn):
+        self.super_fn = super_fn
+        self.trace = []
+
+    def __call__(self, computation):
+        # TODO find a way to peak instead of pop/repush
+        slot = computation.stack_pop1_int()
+        computation.stack_push_int(slot)
+        self.super_fn(computation)
+        self.trace.append(slot)
+
+
+def find_storage_slot(contract, function_selector: str, *args):
```

Comment:
this might be better as just `update_storage_slot`? i think you don't really need to return the slot if you update it as soon as you find it